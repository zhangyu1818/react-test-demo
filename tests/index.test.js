import React from "react";
import { render, fireEvent } from "@testing-library/react";

test("按钮点击", () => {
  const onClick = jest.fn(); // 测试函数
  // render 用来渲染元素
  const { getByLabelText } = render(
    <button aria-label="Button" onClick={onClick} />
  );
  // getByLabelText 可以通过aria-label的值来获取元素
  const btn = getByLabelText("Button");
  fireEvent.click(btn); // 模拟点击事件
  expect(onClick).toBeCalled(); // 期望被调用
  expect(onClick).toBeCalledTimes(1); // 期望被调用一次
});
