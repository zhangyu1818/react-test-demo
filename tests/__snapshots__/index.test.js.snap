// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 测试 1`] = `
<DocumentFragment>
  <form>
    <input
      name="test"
      type="text"
    />
    <button
      type="submit"
    >
      submit
    </button>
  </form>
</DocumentFragment>
`;
